httpfilter.go

```diff
@@ -16,14 +16,14 @@ var NoOpHttpFilter = &api.PassThroughStreamFilter{}
 //
 //	package main
 //	func init() {
-//		RunHttpFilter(new(UserhttpFilter), ConfigOptions{
+//		RunHttpFilter(filterName, filterFactory, ConfigOptions{
 //			BaseConfig: new(UserHttpFilterConfig),
 //		})
 //	}
-func RunHttpFilter(filter HttpFilter, options ConfigOptions) {
+func RunHttpFilter(filterName string, filterFactoryFunc func() HttpFilter, options ConfigOptions) {
```

Comment:
i suggest to introduce a new type instead of `func() HttpFilter`, like `HttpFilterFactory`, wdyt @commoddity ?

Reply:
That sounds like a good idea. I will implement your suggestions and update the PR later today. Thanks for the quick review.

Reply:
Have updated to address this and the other review comments.